### DISCUSSION TITLE
Formulating an ODE system

### DISCUSSION THREAD

**Discussion by n-mourdou** [pybamm-team/PyBaMM#1968] [Q&A]
Hey everyone, 

I have a question regarding the definition of an ODE.

I have a quite big vector as a  (time) dependent variable, so it is difficult to define explicitly its rhs part by hand. 

Is there a way to define a big vector variable and then access certain slices or coordinates of it and define their time derivative?

Thank you

**Comment by valentinsulzer** [comment]
You can maybe use e.g. `StateVector(slice(10:20))` to do what you are describing. But I don't really understand why you are trying to do it like this. Can you post the full model in mathematical notation?
